src/stream-chat-net/AppSettings.cs

```diff
@@ -1,29 +1,125 @@
+        public string WebhookURL { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore, PropertyName = "webhook_events")]
+        public List<string> WebhookEvents { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore, PropertyName = "user_search_disallowed_roles")]
+        public List<string> UserSearchDisallowedRoles { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore, PropertyName = "before_message_send_hook_url")]
+        public string BeforeMessageSendHookUrl { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore, PropertyName = "image_moderation_labels")]
+        public List<string> ImageModerationLabels { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore, PropertyName = "image_moderation_enabled")]
+        public bool? ImageModerationEnabled { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore, PropertyName = "auto_translation_enabled")]
+        public bool? AutoTranslationEnabled { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore, PropertyName = "sqs_url")]
+        public string SqsUrl { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore, PropertyName = "sqs_key")]
+        public string SqsKey { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore, PropertyName = "sqs_secret")]
+        public string SqsSecret { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore, PropertyName = "image_upload_config")]
+        public FileUploadConfig ImageUploadConfig { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore, PropertyName = "file_upload_config")]
+        public FileUploadConfig FileUploadConfig { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore, PropertyName = "revoke_tokens_issued_before")]
+        public DateTime? RevokeTokensIssuedBefore { get; set; }
+
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore, PropertyName = "enforce_unique_usernames")]
+        public UniqueUsernameEnforcementPolicy? EnforceUniqueUsernames { get; set; }
+    }
+
+    public class AppSettings : AppSettingsBase
+    {
 
         [JsonProperty(NullValueHandling = NullValueHandling.Ignore, PropertyName = "apn_config")]
         public APNConfig APNConfig { get; set; }
 
         [JsonProperty(NullValueHandling = NullValueHandling.Ignore, PropertyName = "firebase_config")]
         public FirebaseConfig FirebaseConfig { get; set; }
 
-        [JsonProperty(NullValueHandling = NullValueHandling.Ignore, PropertyName = "webhook_url")]
-        public string WebhookURL { get; set; }
+        [JsonProperty(NullValueHandling = NullValueHandling.Ignore, PropertyName = "huawei_config")]
+        public HuaweiFields HuaweiConfig { get; set; }
```

Comment:
himm there is a difference between Fields and Config 

Fields is from GetApp endpoint to read existing configuration (it only says Enabled or not).

Config is to configure so it takes `id` and `secret`.

All push providers can take `disabled` field too to clear configs.

Reply:
Uhh 🤦 good catch. Fixed it.